Add version 4.8.1 of System.Reflection.DispatchProxy and 6.1.1 of System.Runtime.CompilerServices.Unsafe

carlossanlop opened the pull request:
This PR is result of testing the parallel dependency updates to ensure source-build does not break: https://github.com/dotnet/dotnet/pull/133

System.ValueTuple did not generate anything when running generate.sh against it:
```
./generate.sh --package System.ValueTuple,4.6.0
Restore complete (0.5s)
  PackageSourceGeneratorTask succeeded (3.1s) → artifacts/bin/PackageSourceGeneratorTask/Debug/net10.0/PackageSourceGeneratorTask.dll
  PackageSourceGenerator succeeded (0.2s) → Package source generation skipped for System.ValueTuple (v4.6.0) as it doesn't contain any compile items.

Build succeeded in 4.7s
```

**IMPORTANT**: I need to merge this PR at the same time as all the other PRs in the other repos, to prevent breaking the dependency ingestion in the dotnet/sdk repo.

@MichaelSimons Please remind me if we need to include the feed or not.

MichaelSimons commented:
> @MichaelSimons Please remind me if we need to include the feed or not.

MichaelSimons commented on NuGet.config:
```diff
@@ -5,6 +5,7 @@
   </fallbackPackageFolders>
   <packageSources>
     <clear />
+     <add key="darc-pub-dotnet-maintenance-packages-f9ed1720" value="https://dnceng.pkgs.visualstudio.com/public/_packaging/darc-pub-dotnet-maintenance-packages-f9ed1720/nuget/v3/index.json" />
```
This should be moved to the other darc feed.  It would be helpful to add a comment like the other feed as well.

carlossanlop replied:
Sorry, can you elaborate?

MichaelSimons replied:
Sorry my wording was poor.  Please move it next to the other existing darc feed - https://github.com/dotnet/source-build-reference-packages/blob/main/NuGet.config#L24

carlossanlop replied:
Ah! Sure, will do.

MichaelSimons commented on src/referencePackages/src/system.runtime.compilerservices.unsafe/6.1.1/System.Runtime.CompilerServices.Unsafe.6.1.1.csproj:
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
This project will need to be added to https://github.com/dotnet/source-build-reference-packages/blob/main/eng/Build.props#L45 per the [instructions](https://github.com/dotnet/source-build-reference-packages/blob/main/README.md#workflow)

```
Add DependencyPackageProjects for all new projects in the [eng/Build.props](https://github.com/dotnet/source-build-reference-packages/blob/main/eng/Build.props#L9) in the correct dependency order.
```

MichaelSimons commented:
> System.ValueTuple did not generate anything when running generate.sh against it:

I take it the package isn't referenced in the SB graph given your VMR validation was green.

carlossanlop commented:
Should we merge this before all the other PRs? I think this one would not break the sdk deps flow if we merge it first.

MichaelSimons commented:
> Should we merge this before all the other PRs? I think this one would not break the sdk deps flow if we merge it first.
> 
> Edit: Need some clarification from my team offline. Making it draft for now.

Correct, This is additive so it can be merged ahead of the incoming changes.

MichaelSimons commented:
Once you mark it as ready for review, I will go ahead and merge it.

carlossanlop commented:
Wait, let me update the feed. I had to generate a new version for a new package.

carlossanlop commented:
Okay! Ready. I updated the feed to the latest, then I added System.Runtime.CompilerServices and also added the recently updated System.Reflection.DispatchProxy to the repo and updated build.props. As previously mentioned, System.ValueTuple is not getting added by generate.sh.
